wandb/sdk/data_types/base_types/media.py

```diff
@@ -27,6 +27,14 @@
 def _wb_filename(
     key: Union[str, int], step: Union[str, int], id: Union[str, int], extension: str
 ) -> str:
+    # Avoid writing to absolute paths by striping any leading slashes.
+    key = str(key).lstrip(os.sep)
+
+    # Avoid directory traversal by replacing dots with underscores.
+    keys = key.split(os.sep)
+    keys = [k.replace(".", "_") if k == "." or k == ".." else k for k in keys]
```

Comment:
```suggestion
    keys = [k.replace(".", "_") if k in ("." , "..") else k for k in keys]
```
also i think `os` has `os.pardir` and `os.curdir`, for these. What happens if it has multiple dots, i.e. something like `...` will it cause any issues?

Reply:
nice, thanks

Reply:
what happens with multiple `///` mid name: something like `/whatever///lol/bla`

Reply:
> what happens with multiple /// mid name: something like /whatever///lol/bla

When a file path contains a `/` in the middle then this is ignored. 
I just tested this on with the example you provided `/whatever///lol/bla` and verified that the media files are placed under `media/images/whatever/lol/`

Reply:
now that i think about it, i don't know if it is correct to create nested folders from the key names, seems to me that it should stay under images. 
I think we should keep the key value as is, but only replace the key when it is used to construct the path.

Reply:
> now that i think about it, i don't know if it is correct to create nested folders from the key names, seems to me that it should stay under images.

Do you mean that when we use the key like `/whatever///lol/bla` this should create a file under the images directory
`images/whatever___lol_bla.png`? (as an example)

This would also affect the current behavior for media names without any slashes would function,
if I log an image with the name `whatever/lol/blah` this will create a nested folder as well.

Reply:
yep, but i'm exactly saying, should we even create nested folders based on the names, is this is the expected behavior? do we document it anywhere in our docs or code?

Reply:
I don't see it formally documented anywhere, but this is a side effect of using the key directly in the media filename https://github.com/wandb/wandb/blob/main/wandb/sdk/data_types/base_types/media.py#L125-L128

What is your reasoning for why we should not have nested folders under the images directory?
I can see this being useful from a user perspective when browsing the images in wandb.ai. It give users the ability to logically group images and browse them later in a folder hierarchy structure.

Reply:
hmmm... yeah i think that's a valid point... so let's keep it as is. i'm convinced :)